src/java/htsjdk/samtools/SRAIndex.java

```diff
@@ -0,0 +1,215 @@
+/*===========================================================================
+*
+*                            PUBLIC DOMAIN NOTICE
+*               National Center for Biotechnology Information
+*
+*  This software/database is a "United States Government Work" under the
+*  terms of the United States Copyright Act.  It was written as part of
+*  the author's official duties as a United States Government employee and
+*  thus cannot be copyrighted.  This software/database is freely available
+*  to the public for use. The National Library of Medicine and the U.S.
+*  Government have not placed any restriction on its use or reproduction.
+*
+*  Although all reasonable efforts have been taken to ensure the accuracy
+*  and reliability of the software and data, the NLM and the U.S.
+*  Government do not and cannot warrant the performance or results that
+*  may be obtained by using this software or data. The NLM and the U.S.
+*  Government disclaim all warranties, express or implied, including
+*  warranties of performance, merchantability or fitness for any particular
+*  purpose.
+*
+*  Please cite the author in any work or product based on this material.
+*
+* ===========================================================================
+*
+*/
+
+package htsjdk.samtools;
+
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.List;
+
+/**
+ * Created by andrii.nikitiuk on 9/4/15.
+ */
+public class SRAIndex implements BrowseableBAMIndex {
+    /**
+     * What is the starting bin for each level?
+     */
+    private static final int[] LEVEL_STARTS = {0,1,9,73,585,4681};
+
+    /**
+     * Reports the maximum number of bins that can appear in a BAM file.
+     */
+    private static final int MAX_BINS = 37450;   // =(8^6-1)/7+1
+    /**
+     * First bin number in last level
+     */
+    private static final int SRA_BIN_INDEX_OFFSET = LEVEL_STARTS[LEVEL_STARTS.length - 1];
+
+    private static int SRA_BIN_SIZE = 16 * 1024;
+    private static int SRA_CHUNK_SIZE = 50000;
+
+    private SAMFileHeader header;
+    private SRAIterator.RecordRangleInfo recordRangleInfo;
+
+    public SRAIndex(SAMFileHeader header, SRAIterator.RecordRangleInfo recordRangleInfo) {
+        this.header = header;
+        this.recordRangleInfo = recordRangleInfo;
+    }
+
+    @Override
+    public int getLevelSize(int levelNumber) {
+        if(levelNumber == getNumIndexLevels()-1)
+            return MAX_BINS-LEVEL_STARTS[levelNumber]-1;
+        else
+            return LEVEL_STARTS[levelNumber+1]-LEVEL_STARTS[levelNumber];
+    }
+
+    @Override
+    public int getLevelForBin(Bin bin) {
+        return LEVEL_STARTS.length - 1;
```

Comment:
Added a check for bin number, and if the bin is not from the last level, throw exception